src/main/java/org/apache/commons/io/input/BrokenInputStream.java

```diff
@@ -117,10 +117,10 @@ public int read() throws IOException {
     /**
      * Throws the configured exception.
      *
-     * @throws IOException always thrown
+     * @throws IOException as configured.
      */
     @Override
-    public synchronized void reset() throws IOException {
+    public void reset() throws IOException {
```

Comment:
Keep `synchronized`, there is no reason to break the superclass' specification here.

Reply:
Could you clarify?  I can't see any mention of `synchronized` [in the javadoc](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/io/InputStream.html#reset()).

Reply:
Uh? Javadoc doesn't document `synchronized` keywords?

Reply:
I didn't explain myself very well.  I meant to say I don't see any mention of locking in the javadoc.

I dug a bit deeper and it seems the synchronization of the `reset` method varies by JDK.  Here's a summary of which JDKs mark it as synchronized, sampled from the ones I had to hand:
|   | java.io.InputStream#reset | java.io.Reader#reset |
| - | ---------------------------------- | ---------------------------- |
| openjdk-7 | [:white_check_mark:](https://github.com/openjdk-mirror/jdk7u-jdk/blob/master/src/share/classes/java/io/InputStream.java#L346) | [:x:](https://github.com/openjdk-mirror/jdk7u-jdk/blob/master/src/share/classes/java/io/Reader.java#L248) | 
| corretto-1.8.0_392 | [:white_check_mark:](https://github.com/corretto/corretto-8/blob/8.392.08.1/jdk/src/share/classes/java/io/InputStream.java#L347) | [:x:](https://github.com/corretto/corretto-8/blob/8.392.08.1/jdk/src/share/classes/java/io/Reader.java#L248) |
| temurin-11.0.21 | :white_check_mark: | :x: |
| corretto-20.0.2.10 | [:x:](https://github.com/corretto/corretto-20/blob/20.0.2.10.1/src/java.base/share/classes/java/io/InputStream.java#L737) | [:x:](https://github.com/corretto/corretto-20/blob/20.0.2.10.1/src/java.base/share/classes/java/io/Reader.java#L398) |
| coretto-21.0.1.12 | [:x:](https://github.com/corretto/corretto-21/blob/21.0.1.12.1/src/java.base/share/classes/java/io/InputStream.java#L740) | [:x:](https://github.com/corretto/corretto-21/blob/21.0.1.12.1/src/java.base/share/classes/java/io/Reader.java#L398) |

AFAIK `java.io.Reader#reset` has never been synchronized, so that one is easy.  For `java.io.InputStream#reset`, I guess it should be synchronized to line up with Java 8, since that's the library's targeted version?

Reply:
Well, this PR is about a new feature and should not flip-flop an unrelated implementation detail IMO. Especially when the setting seems to change between vendor.

Reply:
FWIW, I see synchronized on InputStream Temurin 17 but not Temurin 21.